components/script/dom/url.rs

```diff
@@ -133,6 +133,28 @@ impl URL {
         ServoUrl::parse_with_base(parsed_base.as_ref(), &url.0).is_ok()
     }
 
+    /// <https://url.spec.whatwg.org/#dom-url-parse>
+    pub fn Parse(
+        global: &GlobalScope,
+        url: USVString,
+        base: Option<USVString>,
+    ) -> Option<DomRoot<URL>> {
+        // Step 1: Let parsedURL be the result of running the API URL parser on url with base,
+        // if given.
+        let parsed_base = base.map(|b| ServoUrl::parse(b.0.as_str()).ok()).flatten();
```

Comment:
Can you avoid the flatten with `and_then`?

```suggestion
        let parsed_base = base.and_then(|base| ServoUrl::parse(base.0.as_str()).ok())
```